Automattic/wp-calypso
# Editor: Make the AddTerm Component more reusable

Previously the AddTerm Component was tied with the Post. In this PR, I make it independent. Its new role is just to add a new the term. 

I also provide an `onSuccess` callback to potentially add the created term to the edited post (It can be used for anything else).

I had also to update to `AddTermForPost` Action Creator to handle only the post edition and attach the term to the edited post. So, we don't need the post middleware anymore since the two actions are now independent.

I also added the possibility to enter a description for the created term (will be used for the Term Management Page).

A next step about this issue #8584 is to handle editing terms on the same component.

**Testing instructions**

* Create a post
* Add a new category to this post (using the add term modal)
* Make sure the category is attached to the post

cc @timmyc
